### PR TITLE
New Lesson - Robot Framework

### PR DESCRIPTION
**Pull request by lara29**
@Mierdin @cloudtoad Please review the changes.

cc @saimkhan92

### CODE REVIEW

File: images/utility/push.sh [hunk 1]
```diff
@@ -1,2 +1,4 @@
-docker build -t antidotelabs/utility .
-docker push antidotelabs/utility
+#docker build -t antidotelabs/utility .
+#docker push antidotelabs/utility
+docker build -t la29ra/utility .
```
Comment (Mierdin) [inline images/utility/push.sh]: Please undo changes

File: lessons/lesson-29/report.html [hunk 2]
```diff
@@ -0,0 +1,2420 @@
+<!DOCTYPE html>
```
Comment (Mierdin) [inline lessons/lesson-29/report.html]: Is this file necessary?

File: lessons/lesson-29/output.xml [hunk 3]
```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Mierdin) [inline lessons/lesson-29/output.xml]: Is this file necessary?

File: lessons/lesson-29/log.html [hunk 4]
```diff
@@ -0,0 +1,2121 @@
+<!DOCTYPE html>
```
Comment (Mierdin) [inline lessons/lesson-29/log.html]: Is this file necessary?

File: lessons/lesson-29/stage1/guide.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
+# Chapter 1 - Introduction to Robot Framework
+
```
Comment (Mierdin) [inline lessons/lesson-29/stage1/guide.md]: Please give yourself credit here.

File: lessons/lesson-29/stage2/guide.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
+# Chapter 2 - Developing Test cases using Robot framework - Part 1
```
Comment (Mierdin) [inline lessons/lesson-29/stage2/guide.md]: This Part1 / Part2 is confusing against Chapter 2 / Chapter 3. Can you come up with a different name entirely for Chapter 3 instead?

File: lessons/lesson-29/stage1/guide.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+Robot is a `keyword-driven` testing framework, where all the actions and functions of the test suite are performed using simple, human-understandable keywords.
+
+A generic architecture of the Robot framework is shown in the lesson diagram.
+
+Reference: [Robot Framework User Guide](http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#id413)
+
+Let's move on to the various concepts used in Robot Framework
+
+---
```
Comment (Mierdin) [inline lessons/lesson-29/stage1/guide.md]: I notice you're using this in many places across this lesson. Any reason why? Can we remove it?

**Comment by Mierdin** [comment]
Also, note that CI is failing because no CHANGELOG update. Please update the changelog as indicated in other sections.

**Comment by lara29** [comment]
Thanks for the review @Mierdin. We will update the lesson as per your comments.

File: lessons/lesson-29/report.html (hunk 2 above)
Comment (lara29) [reply lessons/lesson-29/report.html]: I'll remove these html files.

File: lessons/lesson-29/stage1/guide.md (hunk 7 above)
Comment (lara29) [reply lessons/lesson-29/stage1/guide.md]: Did you mean the --- ? This is used for dividing the section with a line.

**Comment by saimkhan92** [comment]
> Also, note that CI is failing because no CHANGELOG update. Please update the changelog as indicated in other sections.

CHANGELOG updated

**Comment by saimkhan92** [comment]
All the commits have been squashed. 
All CI checks passing.

File: lessons/lesson-29/stage2/guide.md (hunk 6 above)
Comment (lara29) [reply lessons/lesson-29/stage2/guide.md]: Changed lesson titles

**Comment by lara29** [comment]
@Mierdin We have made all the requested changes. Please review and let us know if we need to make any other modifications. Thanks!
cc @saimkhan92

**Comment by lara29** [comment]
Thanks @Mierdin
